avocado/plugins/sysinfo.py

```diff
@@ -36,13 +37,13 @@ def _init_sysinfo(self, job_logdir):
             self.sysinfo = sysinfo.SysInfo(basedir=basedir)
 
     def pre(self, job):
-        if job.config.get('sysinfo', None) != 'on':
+        if not settings.get('sysinfo.collect', 'enabled', bool):
```

Comment:
My main problem with this approach is that it breaks the Job API use case of allowing users to set dictionaries as their job configuration.  IMO, the processing/merge of configuration file and options should happen at a given point in the application, and after that, *at runtime*, there should be no need to deal with an Settings instance, unless I'm missing something.

Reply:
I think that I understood. I agree with you, and probably we have two options here:

1. Any piece of code that would like to get settings/config values will do this:
```python3
# This is used on this PR
from avocado.core.future import settings
value = settings.get('sysinfo.collect', ' enabled', bool)
```
2. We will pass the settings object to `method()` call instead of an "argparse config". See line 93 on `avocado/core/app.py`. So they can do `self.settings` (or `self.config`, or any good name here).

I choose to go with 1 here to not break the current usage since this is a future experiment. But the idea is: when everything is using this new "future.settings" we can switch to 2 if need.

Makes sense?